onadata/apps/logger/models/attachment.py

```diff
@@ -86,3 +90,16 @@ def file_hash(self):
     def filename(self):
         if self.media_file:
             return os.path.basename(self.media_file.name)
+
+    @classmethod
+    def soft_delete(self, instance, user=None):
+        """
+        Soft deletes an attachment by adding a deleted_at timestamp.
+        """
+        if instance:
+            queryset = self.__class__.objects.filter(
```

Comment:
Did you mean to make this `cls.objects.filter(name__in=instance.get_expected_media())` ?

Reply:
As is makes sense if remove `classmethod` and make this an instance method, which is better if this always requires an instance to be passed, instead operate off of the instance